src/swsssdk/interface.py

```diff
@@ -366,4 +366,4 @@ def _connection_error_handler(self, db_name):
         logger.warning('Could not connect to Redis--waiting before trying again.')
         self.close(db_name)
         time.sleep(self.CONNECT_RETRY_WAIT_TIME)
-        self.connect(db_name, True)
+        self.connect(self.redis_clients[db_name].connection_pool.connection_kwargs['db'], db_name, True)
```

Comment:
>self.redis_clients[db_name] [](start = 21, length = 27)

It is already closed by Line 367. So it is not in the array.

Reply:
close only do a disconnect, the redis obj is still there, I think that's a minor bug, we suppose to remove old obj after close. fix it in update and still using local dict to store db_name & db_id mapping